deps/rabbitmq_mqtt/src/rabbit_mqtt_reader.erl

```diff
@@ -7,146 +7,165 @@
 
 -module(rabbit_mqtt_reader).
 
--behaviour(gen_server2).
+-behaviour(gen_server).
+-behaviour(ranch_protocol).
 
--export([start_link/2]).
+-include_lib("kernel/include/logger.hrl").
+-include_lib("rabbit_common/include/logging.hrl").
+
+-export([start_link/3]).
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
-         code_change/3, terminate/2, handle_pre_hibernate/1]).
+         code_change/3, terminate/2, format_status/1]).
 
--export([conserve_resources/3, start_keepalive/2,
+-export([conserve_resources/3,
          close_connection/2]).
 
--export([ssl_login_name/1]).
 -export([info/2]).
 
--include_lib("amqp_client/include/amqp_client.hrl").
 -include("rabbit_mqtt.hrl").
 
--define(SIMPLE_METRICS, [pid, recv_oct, send_oct, reductions]).
--define(OTHER_METRICS, [recv_cnt, send_cnt, send_pend, garbage_collection, state]).
+-define(HIBERNATE_AFTER, 1000).
+-define(PROTO_FAMILY, 'MQTT').
+
+-record(state,
```

Comment:
I just did it for `rabbit_mqtt_processor` where the number of fields dropped from 26 to 11.
I think I would leave it as-is for `rabbit_mqtt_reader` because it has "only" 13 fields, if that's okay. Splitting also causes access to become more expensive.